pages/p/[id].test.jsx

```diff
@@ -0,0 +1,63 @@
+import '@testing-library/jest-dom'
+import { render, screen } from '@testing-library/react'
+import { createDynamicRouteParser } from "next-router-mock/dynamic-routes";
+import mockRouter from 'next-router-mock';
+import Page from './[id].tsx';
+import { getServerSideProps } from "./[id].tsx"
+
+jest.mock('next/router', () => jest.requireActual('next-router-mock'))
+mockRouter.useParser(createDynamicRouteParser([
+    "/[id]"
+  ]));
+ 
+describe('Page', () => {
+  const params = { id: 1 }
+  const post = {
+    id: '1',
+    title: 'basketball rental',
+    owner: {
+        id: 'clvwz3m1f000014f8co42anoq',
+        name: 'chenxi',
+        email: 'lin.chenxi14@gmail.com'
+    },
+    ownerId: 'clvwz3m1f000014f8co42anoq',
+    createdAt: '2024-05-07T22:49:54.043Z',
+    updatedAt: '2024-05-07T00:00:00.000Z'
+  }
+
+  it('renders a title', async () => {
+  
+    global.prisma.post.findUnique.mockResolvedValue(post)
+    const result = await getServerSideProps(params)
+    mockRouter.push("/pages/p/1")
+    render(<Page {...result.props} />)
+    
+    const headingTitle= screen.getByTestId('heading-title')
+    expect(headingTitle).toBeInTheDocument()
+    expect(headingTitle).toHaveTextContent(result.props.title)
+  })
+
+
+  it('render post name', async() => {
+  
+    global.prisma.post.findUnique.mockResolvedValue(post)
+    const result = await getServerSideProps(params)
+    mockRouter.push("/pages/p/1")
+    render(<Page {...result.props} />)
+    
+    const postName = screen.getByTestId('post-name')
+    expect(postName).toBeInTheDocument()
+    expect(postName).toHaveTextContent(result.props.owner.name)
+  })
+
+  it('fail if props is access wrong', async () => {
+      global.prisma.post.findUnique.mockResolvedValue(post)
+      const result = await getServerSideProps(params)
+      console.log(result.props)
+      mockRouter.push("/pages/p/1")
+      render(<Page {...result.props} />)
+      expect((result) => {
+        return result.owner.name
+      }).toThrow("Cannot read properties of undefined (reading 'owner')")
```

Comment:
This tests for what the props contain manually and does not reflect the behavior of the page. What should be done instead is to pass in malformed/empty props and test for the behaviour that takes place on the page.